tests/unit/string_util_test.cpp

```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/common/string_util.h>
+}
+
+TEST(strstr, basic)
+{
+   const char haystack[] = "haystack", needle[] = "needle", \
+   hay[] = "hay", empty[] = "";
+   ASSERT_EQ(strstr(empty, needle), (char *) NULL);
+   ASSERT_EQ(strstr(haystack, needle), (char *) NULL);
+   ASSERT_EQ(strstr(haystack, hay), haystack);
+}
+
+TEST(strncpy, basic)
+{
+   char dest[4], src[] = "a";
+   strncpy(dest, src,2);
+   ASSERT_EQ(0, strcmp(dest, "a"));
+}
+
+TEST(strncat, basic)
+{
+   char dest[6] = "abc", src[] = "d";
+   strncat(dest, src, 4);
+   ASSERT_EQ(0, strcmp(dest, "abcd"));
+}
+
+TEST(isxdigit, basic)
+{
+   ASSERT_EQ(isxdigit(48), true);
```

Comment:
Same as above. How can we be absolutely sure you’re calling Tilck’s implementation of this standard function? Even if in this specific contex for a series of reasons you might be actually calling Tilck code, this approach is fragile. Prefix all of those functions and add aliases.